src/ann_solo/reader.py

```diff
@@ -231,6 +299,324 @@ def get_version(self) -> str:
+                          re.IGNORECASE)
+        if perMZ:
+            perMZ = re.search('[0-9]+.[0-9]+', perMZ.group(0))
+        else:
+            perMZ = re.search('Parent=\s?[0-9]+.[0-9]+', spectrum_metadata,
+                              re.IGNORECASE)
+            perMZ = re.search('[0-9]+.[0-9]+', perMZ.group(0))
+        ##Retrieve MZ & Intensities
+        file = io.StringIO(spectrum)
+        MZ_Intensity_annot = pd.read_csv(file, sep="\t", header=None)
+
+        if MZ_Intensity_annot.shape[1] > 2:
+            annotation = [self._parse_fragment_annotation(mz, annotation)
+                          for mz, annotation in
+                          zip(MZ_Intensity_annot[0], MZ_Intensity_annot[2])]
+        else:
+            annotation = [None] * len(MZ_Intensity_annot[0])
+        spectrum = MsmsSpectrum(identifier, perMZ.group(0), charge,
+                                MZ_Intensity_annot[0].to_numpy(copy=True),
+                                MZ_Intensity_annot[1].to_numpy(copy=True),
+                                annotation,
+                                is_decoy=decoy)
+
+        spectrum.peptide = peptide
+
+        return spectrum
+
+    # For parallel processing
+    def _parse_batch_sptxt_spectra(self, raw_spectra: List[str], ids: List[
+        int]) -> Iterator[MsmsSpectrum]:
+        """
+        Takes a list of  raw spectra data retrieved from an sptxt file and
+        parses it to a structured object of type MsmsSpectrum.
+
+        Parameters
+        ----------
+        raw_spectrum : List[string]
+            The spectrum in a raw format.
+        identifier : List[int]
+            Incremented identifier of the spectrum in the library.
+
+        Returns
+        -------
+        Iterator[MsmsSpectrum]
+            An iterator of spectra in the given library file.
+        """
+        for spectrum in list(Parallel(n_jobs=self._num_threads)(
+                delayed(self._parse_sptxt_spectrum)(raw_spectrum, id) for
+                raw_spectrum, id in zip(raw_spectra, ids))):
+            yield spectrum
+
+    def parse_sptxt(self) -> Iterator[MsmsSpectrum]:
+        """
+        Open the sptxt spectra library file and parses it
+        to read all spectra.
+
+        Returns
+        -------
+        Iterator[MsmsSpectrum]
+            An iterator of spectra in the given library file.
+
+        """
+        with open(self._filename, 'rb') as file:
+            mmapped_file = mmap.mmap(file.fileno(), 0, access=mmap.ACCESS_READ)
+            raw_spectra_list = []
+            for i, raw_spec in enumerate(re.finditer(
+                    b'(?<![a-zA-Z])Name:\s?(?:(?!((?<![a-zA-Z])Name:\s?)).|\n)*',
+                        mmapped_file.read(),
+                        re.IGNORECASE),
+                    1):
+                raw_spectra_list.append(
+                    '\n'.join(raw_spec.group(0).decode('utf-8').splitlines()))
+                if i % self._num_threads:
```

Comment:
**issue:** Can this be simplified by skipping adding the spectra to a list but just providing the generator to the threaded processing? I think that should be equally efficient (maybe even a bit faster) and is less convoluted code.